### PR TITLE
Feature/pedal info valodation

### PR DESCRIPTION
**Pull request by miggelucas**
## What I Have done

- Knob model for representing name and level for that parameter
- KnobView for displaying knob model info
- HomeView as a pedal list view, searchable
- HomeView empty state
- AddPedalView sheet for adding new pedals
- add new pedals alert in case of missing info

### CODE REVIEW

File: PedalCore/Models/Knob.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//
+
+import Foundation
+
+/// Representation for a pedal knob
+struct Knob {
+    var id: UUID = UUID()
+    
+    /// Knob label
+    var parameter: String
```
Comment (matheusberger) [inline PedalCore/Models/Knob.swift]: I think it's simpler if we call it "name"

File: PedalCore/Models/Pedal.swift [hunk 2]
```diff
@@ -7,8 +7,63 @@
...
+    var id: UUID = UUID()
     var name: String
     var brand: String
-    var knobs: [String: Int]
+    var knobs: [Knob]
+    
+}
+
+
+extension Pedal {
```
Comment (matheusberger) [inline PedalCore/Models/Pedal.swift]: Please move this extension to a new file, so we can exclude it from the official builds.

File: PedalCore/ViewModels/HomeViewModel.swift [hunk 3]
```diff
@@ -8,13 +8,72 @@
...
+    
+    public var filteredPedals: [Pedal] {
+        if searchText.isEmpty {
+            return allPedals
+        } else {
+            return allPedals.filter { pedal in
+                pedal.name.localizedCaseInsensitiveContains(searchText) || pedal.brand.localizedCaseInsensitiveContains(searchText)
+            }
+        }
+
```
Comment (matheusberger) [inline PedalCore/ViewModels/HomeViewModel.swift]:
```suggestion
```

File: PedalCore/ViewModels/HomeViewModel.swift [hunk 4]
```diff
@@ -8,13 +8,72 @@
...
+    }
+
+    func addIconPressed() {
+        isShowingSheet = true
+    }
+    
+    // for debugging
+    func populatePedals() {
+        allPedals = Pedal.pedalSample()
+
```
Comment (matheusberger) [inline PedalCore/ViewModels/HomeViewModel.swift]:
```suggestion
```

File: PedalCore/Views/CreatePedalView.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                        switch pedalError {
+                        case .missingName:
+                           alertMessage = "Please, provide the pedal name"
+                        case .missingBrand:
+                            alertMessage = "Please, provide a brand of the pedal"
+                        case .missingKnobs:
+                            alertMessage = "Please, provide a knob name for your pedal"
+                        }
+                    }
+                }
```
Comment (matheusberger) [inline PedalCore/Views/CreatePedalView.swift]: Please move all logic and error handling into the view's view model

**Review by matheusberger — CHANGES_REQUESTED** [review]
Please remove all trailing empty lines from the code while addressing the other comments.

File: PedalCore/Models/Pedal.swift (hunk 2 above)
Comment (miggelucas) [reply PedalCore/Models/Pedal.swift]: It makes totally sense!

File: PedalCore/Views/CreatePedalView.swift (hunk 5 above)
Comment (miggelucas) [reply PedalCore/Views/CreatePedalView.swift]: Makes senes! Done

**Comment by miggelucas** [comment]
Changes proposed in code review done!

**Review by matheusberger — APPROVED** [review]
LGTM!

**Pull request merged**